desktop/dev-environments.md

```diff
@@ -0,0 +1,193 @@
+
+Dev Environments also allow you to switch between your developer environments or your team members' environments, move between branches to look at changes that are in progress, without moving off your current Git branch.
+
+> **Preview**
+>
+> The Dev Environments feature is currently offered as a Preview. We recommend that you do not use this in production environments.
+
+To access Dev Environments, from the Docker menu, select **Dashboard** > **Dev Environments**.
+
+![Dev environment intro](/images/dev-env-create.png){:width="700px"}
+
+## Prerequisites
+
+Dev Environments are available as part of Docker Desktop 3.5.0 release. Download and install **Docker Desktop 3.5.0** or higher:
+
+- [Mac](https://docs.docker.com/docker-for-mac/release-notes/)
+- [Windows](https://docs.docker.com/docker-for-windows/release-notes/)
+
+To get started with Dev Environments, you must have the following tools and extension installed on your machine:
+
+- [Git](https://git-scm.com){:target="_blank" rel="noopener" class="_"}
+- [Visual Studio Code](https://code.visualstudio.com/){:target="_blank" rel="noopener" class="_"}
+- [Visual Studio Code Remote Containers Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers){:target="_blank" rel="noopener" class="_"}
+
+Click **Install** to download and install any missing tools.
+
+## Start a single container Dev Environment
+
+The simplest way to get started with Dev Environments is to create a new environment by cloning the Git repository of the project you are working on. For example, let us create a new Dev Environment using a simple `singe-dev-env` project from the [Docker Samples](https://github.com/dockersamples/single-dev-env){:target="_blank" rel="noopener" class="_"} GitHub repository.
+
+1. First, let's copy `git@github.com:dockersamples/single-dev-env.git` and add it to the **Create** field on the **Create a Development Environment** page.
+3. Now, click **Create**.
+
+    This clones the Git code inside a volume, determines the best image for your Dev Environment, and finally, opens VS Code inside the Dev Environment container.
+
+![Single container Dev environment](/images/dev-env-container.png){:width="700px"}
+
+In the above example, the names `amazing_mclaren` and `stoic_carver` are randomly generated. You'll most likely see different names when you create your Dev Environment.
+
+Hover over the container and click **Open in VS Code** to start working in VS Code as usual. You can also open a terminal in VS Code, and use Git to push or pull code to your repository, or switch between branches and work as you would normally.
+
+You can launch the application by running the command `make run` in your VS Code terminal. This opens an http server on port 8080. Open [http://localhost:8080](http://localhost:8080) in your browser to see the running application.
+
+![Single container Dev environment](/images/dev-env-localhost.png){:width="700px"}
+
```

Comment:
Maybe emphasis a bit more the magical parts after this first step ?
* the repo was cloned in a container using your git credentials, and does not collide with any of you local files on your host
* you don't need to install any tools, like go tools, all this is setup automatically, and also does not collide with any of you local tooling
* You can run the app server (inside a container) from the VSCode terminal and it is accessible from the host automatically

Reply:
@gtardif PTAL